docs/compatibility.md

```diff
@@ -0,0 +1,77 @@
+# StableHLO Compatibility
+
+StableHLO is a backward compatible ML compute opset inspired by HLO/MHLO.
+This document explains the kind and the extent of the compatibility guarantees
+that StableHLO provides, based on the process established in
+[the compatibility RFC](../rfcs/20220912-compatibility.md).
+
+## Versions
+
+The current version of StableHLO is **0.9.0**.
+
+In the 0.x.x series, StableHLO is providing limited compatibility guarantees
+as described in the Guarantees section. In H2 2023, we are planning to release
+StableHLO 1.0.0, which will provide full compatibility guarantees - see the
+Future work section for more details.
+
+## Guarantees
+
+**1 month of backward compatibility:** VHLO program serialized by an old version
+of libStablehlo has the same semantics* when deserialized by a new version of
+libStablehlo if these versions are built from openxla/stablehlo commits which
+are less than 1 month apart.
+
+**1 month of forward compatibility:** VHLO program serialized by a new version
+of libStablehlo has the same semantics* when deserialized by an old version of
+libStablehlo if these versions are built from openxla/stablehlo commits which
+are less than 1 month apart, unless the program is using new features introduced
+since the old version.
+
+\* VHLO is a dialect that contains versioned snapshots of the StableHLO opset.
+It can be obtained from the StableHLO dialect via
+[--stablehlo-legalize-to-vhlo](../stablehlo/tests/stablehlo_legalize_to_vhlo.mlir),
+and its semantics is defined by [the StableHLO spec](spec.md).
+
+## Out of scope
+
+**Source compatibility** for C, C++ and Python APIs within libStablehlo is
+an aspirational goal. At the moment, we don't offer source compatibility
+guarantees, but please let us know if this is an important use case for you,
```

Comment:
Do we have a ticket for this? Would be a good place to have this discussion

Reply:
Opened #1247.